pandas/_testing/asserters.py

```diff
@@ -6,7 +6,10 @@
 
 import numpy as np
 
-from pandas._libs.lib import no_default
+from pandas._libs.lib import (
+    NoDefault,
```

Comment:
we should probably import from pandas._typing.

alternatively we could have `no_default` as 'Any' in pandas/_libs/lib.pyi and not need to have any reference to it in the function signatures which are visible to end users and could be confusing/misleading. we should never pass `lib.no_default` explicitly unless passing it along to a lower level function